task_processing/interfaces/event.py

```diff
@@ -8,18 +8,36 @@
 
 
 class Event(PRecord):
-    timestamp = field(type=datetime.datetime)
+    type = field(type=str, initial=None)
```

Comment:
you're right that we've not got a good system for serialization, but isn't that a problem to be solved at serialization time? i.e. set a type field in the record written to dynamo and then use that field to create an object of the correct type?

if you feel strongly, then please remove the class hierarchy all together, and encode the type only in the type field, rather than the type field and the class: there isn't anything to say I can't create a ControlEvent with a type 'foo', and confuses which encoding is the source of truth.